client/my-sites/controller.js

```diff
@@ -87,9 +89,11 @@ function createNavigation( context ) {
 		basePath = sectionify( context.pathname );
 	}
 
+	const SidebarComponent = config.isEnabled( 'jetpack-cloud' ) ? JetpackCloudSidebar : Sidebar;
+
 	return (
 		<NavigationComponent
-			path={ context.path }
+			sidebar={ <SidebarComponent path={ context.path } siteBasePath={ basePath } /> }
```

Comment:
I wonder if we could `AsyncLoad` the sidebar component here. Just so we wouldn't import both components when only one of them will be used at any given time. This could have a positive impact on the bundle size, too.

Reply:
Will try this in a different PR.